src/sidebar/services/oauth-auth.js

```diff
@@ -167,9 +167,27 @@ function auth($http, $rootScope, $window, OAuthClient,
     if (!tokenInfoPromise) {
       var cfg = serviceConfig(settings);
 
+      if (cfg && cfg.grantToken === 'postMessage') {
+        tokenInfoPromise = new Promise(function(resolve, reject) {
+          /** Receive postMessage() response with grant token from LMS app. */
+          function receiveMessage(event) {
+            if (event.origin !== "http://localhost:8001") {
+              return;
+            }
+
+            oauthClient().then(function(client) {
+              client.exchangeGrantToken(event.data).then(function(accessToken) {
+                resolve(accessToken)
```

Comment:
I've got the LMS app just sending the grant token directly as the `message` argument to `postMessage` (`event.data` on the receiving end here). In a [previous `postMessage()`-related feature](https://github.com/hypothesis/client/pull/550) we used JSON-RPC format messages with `postMessage()` as the transport layer. We should probably do that here too.